auth/authenticator.hh

```diff
@@ -150,6 +152,9 @@ public:
     ///
     virtual const resource_set& protected_resources() const = 0;
 
+    /// True iff alteration is safe for continuous functioning of this authenticator.
+    virtual bool safe(const cql3::statements::schema_altering_statement&) const { return true; }
+
```

Comment:
This is entangling the cql3 hierarchy into the auth hierarchy (it is already entangled, but in an isolated way, with the table based auth* providers).

It would be better to have an intermediate representation that avoids the entanglement. I'll see later if I can suggest one.